### ISSUE TITLE
Answers Updating

### ISSUE THREAD

**Issue by Aljumaili85** [gunthercox/ChatterBot#1553]
Hi there, 
previously I faced a problem of updating the ChatterBot answers when its could not find the right one for a given question, my chatterbot was trained using data from the ChatterBot dialog corpus .

Then I decide to create my own chatterbot from by creating empty database and inserting the question and the the answer one by one using the example "learning_new_response.py".

the application working like this:
type something to begin ...
me: hi
chatterbot: Is "hi" a coherent response to "hi"?
me: no
chatterbot: please input the correct one
me: Hello Dear!
chatterbot: Responses added to bot!
..............................   and so on until I enter, let say 5 questions and answers. Then if I choose a pre-definde question it will choose a wrong answer.
for example if I write "hi" the chatterbot will respond : Is "I am good thank you." a coherent response to "hi"?
I wondering what is going wrong here! ?

by the way, if I open the created database I can find the all the questions with the right answers but I don't know why it chooses a different answer even though I used the BestMatch as logic adapter.

bellow is my code that has been used:

`from chatterbot import ChatBot
from chatterbot.trainers import ChatterBotCorpusTrainer

# Create a new instance of a ChatBot
bot = ChatBot(
    "Terminal",
    storage_adapter="chatterbot.storage.SQLStorageAdapter",
    trainer = "chatterbot.trainers.ChatterBotCorpusTrainer",
        logic_adapters=[
        {
            "import_path": "chatterbot.logic.BestMatch",
            "statement_comparison_function": "chatterbot.comparisons.levenshtein_distance",
            "response_selection_method": "chatterbot.response_selection.get_most_frequent_response"
        }
    ]
)

#bot.train("chatterbot.corpus.english.conversations")
def get_feedback():

    text = input()

    if 'yes' in text.lower():
        return False
    elif 'no' in text.lower():
        return True
    else:
        print('Please type either "Yes" or "No"')
        return get_feedback()

print("Type something to begin...")

# The following loop will execute each time the user enters input
while True:
    try:
        request=input()
        input_statement = bot.input.process_input(request)
        temp1,response = bot.generate_response(
            input_statement,1
        )

        bot.output.process_response(response)
        print('\n Is "{}" a coherent response to "{}"? \n'.format(response, input_statement))
        if get_feedback():
            print("please input the correct one")
            request1=input()
            response1 = bot.input.process_input(request1)
            bot.learn_response(response1, input_statement)
            print("Responses added to bot!")

    # Press ctrl-c or ctrl-d on the keyboard to exit
    except (KeyboardInterrupt, EOFError, SystemExit):
        break
`

**Comment by Aljumaili85** [comment]
No, actually  I have one and only one answer for each question, but it seems that the BestMatch is not selecting the right matched question , but at the same time BestMatch works good with the database of  dialog corpus

**Comment by gunthercox** [comment]
Interesting, I'll mark this as a possible bug then. I'm not aware of any issues currently effecting how the BestMatch adapter selects responses, but I'd happy to work on a solution if the problem can be identified.

**Comment by Dulce06** [comment]
Hi @gunthercox @Aljumaili85 

I have tried this code but got an error 

**

> input_statement = bot.input.process_input(request)
> AttributeError: 'ChatBot' object has no attribute 'input'.

**

Can you please help me how to get rid of it.
Many thanks

**Comment by Aljumaili85** [comment]
It a problem of incompatibility of the given code and the installed ChatterBot version. try to install the latest ChatterBot and try again.  meanwhile i will try to see whats wrong

**Comment by Dulce06** [comment]
Hi @Aljumaili85 

Thanks for the prompt response.

Upgraded the chatterbot version. 
I have the following errors now:



> DEFAULT_SESSION_ID = bot.default_session.id
> AttributeError: 'ChatBot' object has no attribute 'default_session'



> Cannot find reference 'input_function' in 'utils.py' less... (Ctrl+F1) 
> Inspection info: This inspection detects names that should resolve but don't. Due to dynamic dispatch and duck typing, this is possible in a limited but useful number of cases. Top-level and class-level items are supported better than instance items.
